### DISCUSSION TITLE
Hack the Xbox Version of the Toypad

### DISCUSSION THREAD

**Discussion by tomwendel** [Wendelpunkt/ToypadChallenge#4] [Ideas]
Since all the existing libraries to control the toypad only supports the non-xbox version, it would be awesome to find someone who is familiar with some usb sniffing to hack the xbox based protocol.

**Comment by pherting** [comment]
Whats the different? I dont Check why the Microsoft Xbox Version doesnt run.. 😅 has it different Firmware or whats the problem exacatly? One the PS Buzzer Controllers are also differences between cable and cablesless but only on some small Parameters and uid.

**Reply by EldoranDev** [reply]
Microsoft at some point decided that all USB peripherals need to implement an extra layer of security (most likely to restrict the use of aftermarket devices etc.). This causes the USB Devices that are used for the XBOX to use a different protocol as it uses an encrypted version of it.

There are attempts (I think even successful ones) to reverse engineer the security algorithm (which I'll not link here for obvious reasons 🙈) but I am not sure if those work the way we need it or if they allow you attach devices to an xbox.

**Comment by tomwendel** [comment]
As discussed yesterday in Jans stream (https://www.twitch.tv/videos/2227684893) there are two options to deal with this:
a) try to sniff the communication between the pad and the xbox to hack that protocol, too. But if EldoranDev is right, this could be a bit more difficult than expected
b) Compare the hardware of xbox and non-xbox. If this is equal we could just flash the driver chip with the unencrypted version (as far as I understood).

**Reply by Tivertoni** [reply]
@tomwendel The only difference between Xbox Toypads and generic Toypads is that the generic ones have an area on the PCB where components have been desoldered. My guess is that these components store the encryption key or are otherwise involved in authentication to an XBOX.

**Reply by Tivertoni** [reply]
![image](https://github.com/user-attachments/assets/c2f726a2-be17-42ef-a1b1-7ac66cd8bc6a)

**Reply by Tivertoni** [reply]
Q19 is most likely a MOSFET, the rest of the pcb uses this one:
[LWW 4d.pdf](https://github.com/user-attachments/files/20536583/LWW.4d.pdf)

**Reply by Tivertoni** [reply]
Also, on the top right side of the main PCB, there is a desoldered resistor at C7.

**Reply by tomwendel** [reply]
Hey Jonas. Thanks for your work! Looks like there are a lots of parts missing in that area. You think removing this parts will disable everything?

Not sure if this topic is still interesting for anybody since I still have a lot of the generic pads in my garage. :D

**Reply by Tivertoni** [reply]
@tomwendel I tried to identify more parts of the PCB, but I couldn’t find any datasheets or public information about them. My best guess is that if these components are removed from an Xbox Toy Pad PCB, it might behave like a generic one, just with a different VID and PID. I haven’t tested this and can’t, as I don’t own an Xbox Toy Pad—or even an Xbox.

**Reply by Tivertoni** [reply]
@tomwendel Also what I noticed, newer versions of the toypad have a different hex inverter(HEF4049B) and the labels on the chips are harder to read, the unit that i tested it on was 60715, and the newer unit I inspected was 42715(imprinted on the back)

**Reply by Tivertoni** [reply]
@tomwendel On the older version of the toypad(both generic) C9 is still soldered in.
